packages/components/src/Typography.astro

```diff
@@ -0,0 +1,118 @@
+---
+interface Props {
+  variant:
+    | "h1"
+    | "h2"
+    | "h3"
+    | "h4"
+    | "h5"
+    | "h6"
+    | "subtitle1"
+    | "subtitle2"
+    | "body1"
+    | "body2"
+    | "caption"
+    | "overline";
+  override?:
+    | "h1"
+    | "h2"
+    | "h3"
+    | "h4"
+    | "h5"
+    | "h6"
+    | "subtitle1"
+    | "subtitle2"
+    | "body1"
+    | "body2"
+    | "caption"
+    | "overline";
+  class?: string;
+}
+
+const elementMap = {
+  h1: "h1",
+  h2: "h2",
+  h3: "h3",
+  h4: "h4",
+  h5: "h5",
+  h6: "h6",
+  subtitle1: "h6",
+  subtitle2: "h6",
+  body1: "p",
+  body2: "p",
+  caption: "p",
+  overline: "p",
+};
+
+const { class: className, variant, override, ...props } = Astro.props;
+
+const Element = override ? elementMap[override] : elementMap[variant];
+---
+
+<Element {...props} class:list={["gx-typography", variant, className]}>
+  <slot />
+</Element>
+
+<style lang="scss">
+  .gx-typography {
+    &.h1 {
```

Comment:
```suggestion
    &.gx-typography-h1 {
```

If we're still planning to move to global styles we should prefix any classes we use to prevent name collisions with user defined classes.